lib/Entity/Schedule.php

```diff
@@ -659,6 +667,11 @@ public function delete()
         $this->displayGroups = [];
         $this->unlinkDisplayGroups();
 
+        // Delete schedule exclusions
+        $scheduleExclusions = $this->scheduleExclusionFactory->query(null, ['eventId' => $this->eventId]);
```

Comment:
Sorry i'm not clear on something - the $exclusion->delete method deletes by `eventId` which I think will delete all of the events returned by the query in one go?

So the first delete will do something, and then the others wont effect any records, because they are already gone.